test/libmpv_test.c

```diff
@@ -15,15 +15,251 @@
  * License along with mpv.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#include <inttypes.h>
 #include <libmpv/client.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
 
-int main(void)
+// Dummy values for the test.
+const char *str = "string";
+int flag = 1;
+int64_t int_ = 20;
+double double_ = 1.5;
+
+// Global handle.
+mpv_handle *ctx;
+
+static inline void fail(void)
```

Comment:
Pretty much every single `fail()` call is also printing shit. I'd make `fail` accept a format string (and print the errors to `stderr` while at it).

```c
static void fail(const char *fmt, ...)
{
	if (fmt) {
		va_list va;
		va_start(va, fmt);
		vfprintf(stderr, fmt, va);
		va_end(va);
	}
	mpv_destroy(ctx);
	exit(1);
}
```

And maybe throw in the format-printf attribute (`PRINTF_ATTRIBUTE` in `osdep/compiler.h`) in there as well along with noreturn attribute too.

Reply:
Good idea.